crates/cast/src/lib.rs

```diff
@@ -494,6 +494,48 @@ where
         Ok(self.provider.get_transaction_count(who).block_id(block.unwrap_or_default()).await?)
     }
 
+    /// #Example
+    /// 
+    /// ```
+    /// use alloy_primitives::Address;
+    /// use alloy_provider::{network::AnyNetwork, ProviderBuilder, RootProvider};
+    /// use cast::Cast;
+    /// use std::str::FromStr;
+    /// 
+    /// # async fn foo() -> eyre::Result<()> {
+    /// let provider =
+    ///     ProviderBuilder::<_, _, AnyNetwork>::default().on_builtin("http://localhost:8545").await?;
+    /// let cast = Cast::new(provider);
+    /// let addr = Address::from_str("0x7eD52863829AB99354F3a0503A622e82AcD5F7d3")?;
+    /// let codehash = cast.codehash(addr, None).await?;
+    /// println!("{}", codehash);
+    /// # Ok(())
+    /// # }
+    pub async fn codehash(&self, who: Address, block: Option<BlockId>) -> Result<String> {
+        Ok(self.provider.get_account(who).block_id(block.unwrap_or_default()).await?.code_hash.to_string())
```

Comment:
afaik `eth_getAccount` is pretty new and is not supported by eg alchemy, should we consider falling back to `eth_getProof`?

Reply:
Sure!